### PR TITLE
Feature/1316 indexes for mongodb performance

### LINKED ISSUE #1261: Introduce more MongbDB indexes
## Background
Currently, many queries to MongoDB fo full collection scans. For big collections such as `run` it creates a lot of pressure for the MongoDB cluster.

## Feature
Investigate production logs and find bottlenecks. Check if corresponding queries can be optimized or new indexes can be added to ease the pressure on MongoDB.

### PR DESCRIPTION
**Pull request by AdrianOlosutean**
The proposed created indexes
Closes #1261

### CODE REVIEW

**Comment by AdrianOlosutean** [comment]
@DzMakatun Do you think there are other indexes which can be added?

File: migrations/src/main/scala/za/co/absa/enceladus/migrations/migrations/MigrationToV1.scala [hunk 1]
```diff
@@ -39,6 +39,8 @@ object MigrationToV1 extends MigrationBase with CollectionMigration with JsonMig
   createIndex("dataset", Seq(IndexField("name", ASC), IndexField("version", ASC)), unique = true)
   createIndex("schema", Seq(IndexField("name", ASC), IndexField("version", ASC)), unique = true)
   createIndex("mapping_table", Seq(IndexField("name", ASC), IndexField("version", ASC)), unique = true)
+
+  createIndex("run", Seq(IndexField("dataset", ASC), IndexField("datasetVersion", ASC)))
```
Comment (benedeki) [inline migrations/src/main/scala/za/co/absa/enceladus/migrations/migrations/MigrationToV1.scala:43]: I am not sure how MongoDB work, but isn't this index a subset/sub-index of the one right bellow? (with just little smaller selectivity). 🤔

Comment (benedeki) [reply migrations/src/main/scala/za/co/absa/enceladus/migrations/migrations/MigrationToV1.scala:43]: Yup, according to this [article](https://engineering.mixmax.com/blog/improving-mongo-performance-by-managing-indexes), the new one is redundant and MongoDB engine should be able to use the other index whenever the new indexed might seem useful.

Comment (dk1844) [reply migrations/src/main/scala/za/co/absa/enceladus/migrations/migrations/MigrationToV1.scala:43]: Aaaah, I did not realize it, but it actually makes sense 😊 .

**Pull request closed without merging**